sklearn/utils/tests/test_extmath.py

```diff
@@ -329,6 +330,32 @@ def test_randomized_svd_sign_flip():
         assert_almost_equal(np.dot(u2.T, u2), np.eye(2))
         assert_almost_equal(np.dot(v2.T, v2), np.eye(2))
 
+def test_randomized_svd_sign_flip_with_transpose():
+    # Check if flipping is always done based on u
+    def max_loading_is_positive(u, v):
+        # u_based
+        return((np.abs(u).max(axis=0) == u.max(axis=0)).all(),
+        # v_based
+        	   (np.abs(v).max(axis=1) == v.max(axis=1)).all())
+
+    np.random.seed(0)
+    mat = np.arange(10 * 8).reshape(10, -1)
+
+    # Without transpose
+    u_flipped, _, v_flipped = randomized_svd(mat, 3, flip_sign=True)
+    u_based, v_based = max_loading_is_positive(u_flipped, v_flipped)
+    assert_true(u_based)
+    assert_false(v_based)
+
+    # With transpose
+    u_flipped_with_transpose, _, v_flipped_with_transpose = randomized_svd(
+                                            mat, 3, flip_sign=True,
```

Comment:
you have indentation mis-alignment here too . Setting up flake8 checks in your editor would help spotting those too.